lib/src/main/java/com/diffplug/spotless/FormatterProperties.java

```diff
@@ -0,0 +1,303 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.diffplug.spotless;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Properties;
+import java.util.Set;
+import java.util.logging.Logger;
+import java.util.stream.Collectors;
+
+import javax.annotation.Nullable;
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.w3c.dom.NamedNodeMap;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/** Utility manages settings of formatter configured by properties. */
+public class FormatterProperties {
+
+	private static final Logger LOG = Logger.getLogger(FormatterProperties.class.getName());
+
+	private final Properties properties;
+
+	/**
+	 * Per default the property list is empty and the formatters default
+	 * settings are used.
+	 */
+	public FormatterProperties() {
+		properties = new Properties();
+	}
+
+	/**
+	 * Import settings from a sequence of files (file import is the given order)
+	 *
+	 * @param settingsFiles
+	 *            Sequence of files
+	 * @throws IllegalArgumentException
+	 *            In case the import of a file fails
+	 */
+	public void add(final Iterable<File> settingsFiles) throws IllegalArgumentException {
+		settingsFiles.forEach(file -> add(file));
+	}
+
+	/**
+	 * Import settings from given file. New settings (with the same ID/key)
+	 * override existing once.
+	 *
+	 * @param settingsFile
+	 *            File
+	 * @throws IllegalArgumentException
+	 *            In case the import of the file fails
+	 */
+	public void add(final File settingsFile) throws IllegalArgumentException {
+		if (!(settingsFile.isFile() || settingsFile.canRead())) {
+			String msg = String.format("Settings file '%s' does not exist or can not be read.", settingsFile);
+			throw new IllegalArgumentException(msg);
+		}
+		try {
+			Properties newSettings = FileParser.parse(settingsFile);
+			properties.putAll(newSettings);
+		} catch (IOException | IllegalArgumentException | NullPointerException exception) {
+			StringBuffer message = new StringBuffer(
+					String.format("Failed to add properties from '%s' to formatter settings.", settingsFile));
+			String detailedMessage = exception.getMessage();
+			if (null != detailedMessage) {
+				message.append(String.format(" %s", detailedMessage));
+			}
```

Comment:
I don't think you need to use `StringBuffer` here (or even `StringBuilder`) - just concatenate strings with `+`, the compiler will optimise them as a `StringBuilder` construction (since there's no looping going on).